core-web/libs/contenttype-fields/src/lib/fields/binary-field/components/dot-binary-field-url-mode/dot-binary-field-url-mode.component.ts

```diff
@@ -52,29 +53,32 @@ export class DotBinaryFieldUrlModeComponent implements OnInit, OnDestroy {
     @Output() tempFileUploaded: EventEmitter<DotCMSTempFile> = new EventEmitter<DotCMSTempFile>();
     @Output() cancel: EventEmitter<void> = new EventEmitter<void>();
 
-    private readonly destroy$ = new Subject<void>();
+    private readonly store = inject(DotBinaryFieldUrlModeStore);
+
+    // Form
     private readonly validators = [
         Validators.required,
         Validators.pattern(/^(ftp|http|https):\/\/[^ "]+$/)
     ];
-
-    readonly invalidError = 'dot.binary.field.action.import.from.url.error.message';
-    readonly vm$ = this.store.vm$.pipe(tap(({ isLoading }) => this.toggleForm(isLoading)));
-    readonly tempFileChanged$ = this.store.tempFile$;
     readonly form = new FormGroup({
         url: new FormControl('', this.validators)
     });
 
-    private abortController: AbortController;
+    // Observables
+    readonly vm$ = this.store.vm$.pipe(tap(({ isLoading }) => this.toggleForm(isLoading)));
+    readonly tempFileChanged$ = this.store.tempFile$;
 
-    constructor(private readonly store: DotBinaryFieldUrlModeStore) {
-        this.tempFileChanged$.pipe(takeUntil(this.destroy$)).subscribe((tempFile) => {
-            this.tempFileUploaded.emit(tempFile);
-        });
-    }
+    private readonly destroy$ = new Subject<void>();
+    private abortController: AbortController;
 
     ngOnInit(): void {
         this.store.setMaxFileSize(this.maxFileSize);
+        this.tempFileChanged$
+            .pipe(takeUntil(this.destroy$))
+            .pipe(filter((tempFile) => tempFile !== null))
```

Comment:
why 2 pipes?

Reply:
For fun! Two Pipes are better than one 🧠.

P.S.: I missed removing one.